src/generator/AutoRest.Go/Model/PrimaryTypeGo.cs

```diff
@@ -0,0 +1,147 @@
+            };
+        }
+
+        public PrimaryTypeGo(KnownPrimaryType primaryType) : base(primaryType)
+        {
+            Name.OnGet += v =>
+            {
+                return ImplementationName;
+            };
+        }
+
+        /// <summary>
+        /// Add imports for primary type.
+        /// </summary>
+        /// <param name="imports"></param>
+        public void AddImports(HashSet<string> imports)
+        {
+            if (Import != string.Empty)
+                imports.Add(Import);
+        }
+
+        public virtual string Import
+        {
+            get
+            {
+                switch (KnownPrimaryType)
+                {
+                    case KnownPrimaryType.Date:
+                        return "github.com/Azure/go-autorest/autorest/date";
+                    case KnownPrimaryType.DateTimeRfc1123:
+                        return "github.com/Azure/go-autorest/autorest/date";
+                    case KnownPrimaryType.DateTime:
+                        return "github.com/Azure/go-autorest/autorest/date";
+                    case KnownPrimaryType.Decimal:
+                        return "github.com/shopspring/decimal";
+                    case KnownPrimaryType.Stream:
+                        return "io";
+                    case KnownPrimaryType.Uuid:
+                        return "github.com/satori/uuid";
+                    default:
+                        return string.Empty;
+                }
+            }
+        }
+
+        public virtual string ImplementationName
+        {
+            get
+            {
+                switch (KnownPrimaryType)
+                {
+                    case KnownPrimaryType.Base64Url:
+                        // TODO: add support
+                        return "string";
+
+                    case KnownPrimaryType.ByteArray:
+                        return "[]byte";
+
+                    case KnownPrimaryType.Boolean:
+                        return "bool";
+
+                    case KnownPrimaryType.Date:
+                        return "date.Date";
+
+                    case KnownPrimaryType.DateTime:
+                        return "date.Time";
+
+                    case KnownPrimaryType.DateTimeRfc1123:
+                        return "date.TimeRFC1123";
+
+                    case KnownPrimaryType.Double:
+                        return "float64";
+
+                    case KnownPrimaryType.Decimal:
+                        return "decimal.Decimal";
+
+                    case KnownPrimaryType.Int:
+                        return "int32";
+
+                    case KnownPrimaryType.Long:
+                        return "int64";
+
+                    case KnownPrimaryType.Stream:
+                        return "io.ReadCloser";
+
+                    case KnownPrimaryType.String:
+                        return "string";
+
+                    case KnownPrimaryType.TimeSpan:
+                        return "string";
```

Comment:
I wonder why we are using `string` instead of `time.Duration`. Again, not to fix in this PR, but another moment